src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMismatchAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Descriptors.PX1096_PXOverrideMustMatchSignature);
+
+		public override bool ShouldAnalyze(PXContext pxContext, PXGraphSemanticModel graph) =>
+			base.ShouldAnalyze(pxContext, graph) && graph.Type == GraphType.PXGraphExtension;
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphSemanticModel pxGraphExtension)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var allMembers = pxGraphExtension.Symbol.GetMembers();
+
+			if (!allMembers.IsDefaultOrEmpty)
+			{
+				var methodsWithPxOverrideAttribute = allMembers
+					.OfType<IMethodSymbol>()
+					.Where(m => !m.IsStatic && !m.IsGenericMethod && m.HasAttribute(pxContext.AttributeTypes.PXOverrideAttribute, false))
+					.ToList();
+
+				if (methodsWithPxOverrideAttribute.Any())
+				{
+					var allBaseTypes = pxGraphExtension.Symbol
+						.GetGraphExtensionWithBaseExtensions(pxContext, SortDirection.Ascending, true)
+						.OfType<INamedTypeSymbol>()
+						.Where(s => !s.Equals(pxGraphExtension.Symbol));
+
+					methodsWithPxOverrideAttribute.ForEach(m => AnalyzeMethod(context, pxContext, allBaseTypes, m!));
+				}
+			}
+		}
+
+		private void AnalyzeMethod(SymbolAnalysisContext context, PXContext pxContext, IEnumerable<INamedTypeSymbol> allBaseTypes, IMethodSymbol methodSymbol)
+		{
+			// Here we know that the method is not static and has the correct attribute.
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			foreach (var baseType in allBaseTypes)
+			{
+				var allMembers = baseType.GetMembers(methodSymbol.Name);
+
+				if (!allMembers.IsDefaultOrEmpty)
+				{
+					if (allMembers.OfType<IMethodSymbol>().Any(m => PXOverrideHelper.IsSuitable(methodSymbol, m)))
+					{
+						return;
+					}
+				}
+			}
+
+			var location = methodSymbol.Locations.FirstOrDefault();
+
+			if (location != null)
+			{
+				var diagnostic = Diagnostic.Create(Descriptors.PX1096_PXOverrideMustMatchSignature, location);
+
+				context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+			}
+		}
+
+		private class PXOverrideHelper
+		{
+			/// <summary>
+			/// Special signature check between a derived method with the PXOverride attribute and the base method.
+			/// </summary>
+			/// <param name="baseMethod">The method from the derived class, with the PXOverride attribute</param>
+			/// <param name="pxOverrideMethod">The method from the base</param>
+			/// <returns></returns>
+			public static bool IsSuitable(IMethodSymbol baseMethod, IMethodSymbol pxOverrideMethod)
+			{
+				var methodsCompatibility = GetMethodsCompatibility(baseMethod.Parameters.Length, pxOverrideMethod.Parameters.Length);
+
+				if (methodsCompatibility == MethodsCompatibility.NotCompatible)
+				{
+					return false;
+				}
+
+				if (!pxOverrideMethod.CanBeOverriden() || !pxOverrideMethod.IsAccessibleOutsideOfAssembly())
```

Comment:
I believe this should be the check for the `baseMethod`
```suggestion
				if (!baseMethod.CanBeOverriden() || !baseMethod.IsAccessibleOutsideOfAssembly())
```

Reply:
Thank you for your feedback. This is fixed.